Source/Diagnostics/ReducedDiags/FieldReduction.H

```diff
@@ -0,0 +1,184 @@
+/* Copyright 2021 Neil Zaim
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#ifndef WARPX_DIAGNOSTICS_REDUCEDDIAGS_FIELDREDUCTION_H_
+#define WARPX_DIAGNOSTICS_REDUCEDDIAGS_FIELDREDUCTION_H_
+
+#include "ReducedDiags.H"
+#include "WarpX.H"
+#include "Utils/CoarsenIO.H"
+
+/**
+ *  This class contains a function that computes an arbitrary reduction of the fields. The function
+ *  used in the reduction is defined by an input file parser expression and the reduction operation
+ *  can be either Maximum, Minimum, or Sum.
+ */
+class FieldReduction : public ReducedDiags
+{
+public:
+
+    /** constructor
+     *  @param[in] rd_name reduced diags names */
+    FieldReduction(std::string rd_name);
+
+    /** This function is called at everytime step, and if necessary calls the templated function
+     *  ComputeFieldReduction(), which does the actual reduction computation.
+     */
+    virtual void ComputeDiags(int step) override final;
+
+private:
+    /// Parser to read expression to be reduced from the input file.
+    /// 9 elements are x, y, z, Ex, Ey, Ez, Bx, By, Bz
+    static constexpr int m_nvars = 9;
+    std::unique_ptr<ParserWrapper<m_nvars>> m_parser;
+
+    // Type of reduction (e.g. Maximum, Minimum or Sum)
+    int m_reduction_type;
+
+public:
+
+    /** This function does the actual reduction computation. The fields are first interpolated on
```

Comment:
```suggestion
    /** 
     * This function does the actual reduction computation. The fields are first interpolated on
```
I guess that's just a matter of style, but in most cases doxygen comments start with a `/**` followed by a blank line.